### PR TITLE
Highlight candidate application edits on manage (with activity log fix)

### PR DESCRIPTION
**Pull request by inulty-dfe**
## Context
 
Un reverts https://github.com/DFE-Digital/apply-for-teacher-training/pull/8695

Additional changes can be inspected from b0b2e61807af0e22b1ed993fc0ab1379964ed3e1

ActivityLog in the provider interface collates all audits on applications for the current provider. This query requires access to the application choice. Audits that are created because an application form changes do not have a direct relationship to an application choice so the view logic has no way of generating a link to the relevant application choice.

## Screenshots
|Activity Log|Timeline|
|---|---|
|Screenshot from 2023-10-23 13-48-56|Screenshot from 2023-10-23 13-49-10|
## Changes proposed in this pull request

Here we update the `ActivityLog` query to return the `application_choice_id` column. We can access this when the `Audit#auditable` or `Audit#associated` is not an `ApplicationChoice`

Also added is `ApplicationForm::ColumnSectionMapping`.
This provides a simple API for getting the columns belonging to a section, and getting the section that a column belongs to.

## Guidance to review

Have I collected all the correct columns corresponding to the sections?

## Link to Trello card

[Trello Ticket](https://trello.com/c/OYrhZdB9/858-ca-highlight-candidate-application-edits-on-manage-continued)

### CODE REVIEW

File: app/components/provider_interface/activity_log/application_form.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  module ActivityLog
+    class ApplicationForm
+      attr_reader :event, :section
+      def initialize(event:)
+        @event = event
+        @section = ::ApplicationForm::ColumnSectionMapping.by_column(event.audit.audited_changes.keys.first)
+      end
+
+      def event_description
+        "#{section.humanize} edited"
```
Comment (tomas-stefano) [inline app/components/provider_interface/activity_log/application_form.rb:11]: Can we move to a locale like the suggestion above? e.g `provider_interface.activity_log.application_form.section`

File: app/components/provider_interface/activity_log/interview.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+        @event = event
+        @application_choice = @event.audit.associated
+      end
+
+      def event_description
+        user = event.user_full_name
+        candidate = event.candidate_full_name
+
+        return "#{user} set up an interview with #{candidate}" if event.audit.action == 'create'
+        return "#{user} cancelled interview with #{candidate}" if event.audit.audited_changes.key?('cancelled_at')
```
Comment (tomas-stefano) [inline app/components/provider_interface/activity_log/interview.rb:15]: Can we move to a locale like the suggestion above? e.g `provider_interface.activity_log.interview.section.create` and `provider_interface.activity_log.interview.section.cancelled`

File: app/models/application_form.rb [hunk 3]
```diff
@@ -563,6 +563,73 @@ def continuous_applications?
...
+        # Disability
+        'disability_disclosure' => ['disability_disclosure'],
+      })
+
+      Array(sections).flat_map do |section|
+        mapping[section]
+      end.compact
+    end
+    module_function :by_column, :by_section
+  end
```
Comment (tomas-stefano) [inline app/models/application_form.rb:631]: Can this module lives in its own file outside of ApplicationForm?

**Review by tomas-stefano — COMMENTED** [review]
Excellent work. ✌️ Let me know your thoughts ✌️

File: app/models/application_form.rb (hunk 3 above)
Comment (inulty-dfe) [reply app/models/application_form.rb:631]: I moved it to `ApplicationForm::ColumnSectionMapping` `app/models/application_form/column_section_mapping.rb`

File: app/components/provider_interface/activity_log/application_choice.rb [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+      def initialize(event:)
+        @event = event
+        @application_choice = @event.audit.auditable
+      end
+
+      def event_description
+        user = event.user_full_name
+        candidate = event.candidate_full_name
+
+        case event.application_status_at_event
```
[the remark this answers is not in the document]
Comment (inulty-dfe) [reply app/components/provider_interface/activity_log/application_choice.rb]: Not sure I've implemented it the way you described but I've made it less painful to look at and hopefully easier to understand. Translations included

**Review by tomas-stefano — APPROVED** [review]
✌️ 🚀

**Pull request merged**